### ISSUE TITLE
Rewarded Video Ad did not call delegate methods

### ISSUE THREAD

**Issue by alexeyvolkov-dev** [mopub/mopub-ios-sdk#342]
When closing Rewarded Video by tapping on X (close) button non of "rewardedVideoAdWillDisappear, rewardedVideoAdDidDisappear" and similar delegate methods are called.

- [X ] I am submitting a bug report for existing functionality
- [X ] I visited https://developers.mopub.com/ and found no answer
- [X ] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [X ] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version:
5.14.0

#### Device model and OS Version:
iOS 14.0 (Simulator, iPhone 11 Pro MAX, iPhone X)

#### Xcode Version:
Xcode 12.0

#### Ad Unit IDs used in reproducing the issue:
98c29e015e7346bd9c380b1467b33850 (test one)
90d1acfcdb0f43e5b9c2924bbe895eed (real one)

#### Steps to reproduce the behaviour:
The code was unmodified from version 5.13.1 (all works fine),
Then updated to 5.14.0 (issue appeared)

These two lines for loading the AD:

 ```

...
MPRewardedVideo.loadAd(withAdUnitID: AdManager.rewardLivesAdUnit, withMediationSettings: [])
MPRewardedVideo.setDelegate(self, forAdUnitId: AdManager.rewardLivesAdUnit)
...

func rewardedVideoAdDidLoad(forAdUnitID adUnitID: String!) {
      let rewardVideo = MPRewardedVideo.selectedReward(forAdUnitID: AdManager.rewardLivesAdUnit)
      MPRewardedVideo.presentAd(forAdUnitID: AdManager.rewardLivesAdUnit, from: self, with: rewardVideo)
}

//This method is not called
func rewardedVideoAdWillDisappear(forAdUnitID adUnitID: String!) {
        
      print("Rewarded video will dissapear for adUnit: \(String(describing: adUnitID))")
}
 
```

#### Expected behavior:
When i tap on X (close) button, the SDK should call the delegate methods, will/did disappear 

#### Observed behavior:
None of these delegates is called.

**Comment by hmertkahraman** [comment]
Hi @alexeyvolkov-dev ,

Thanks for reaching out and providing the details. We will check your code, test this internally and get back to you.

MoPub Team

**Comment by IvanoBilenchi** [comment]
I have a very similar issue, but delegate methods not being called are those of fullscreen interstitials (e.g. interstitialDidDisappear: and interstitialWillDisappear:).

**Comment by IvanoBilenchi** [comment]
@hmertkahraman Just a heads up: I had a look at the code, and the issue seems to be caused by `MPFullScreenAdViewController`'s `appearanceDelegate` being deallocated too early, before `viewWillDisappear` and `viewDidDisappear` are called. As a workaround, it is enough to change the `appearanceDelegate` property from `weak` to `strong` and the delegate methods start being called again.

I haven't looked at the code in such detail as to rule out this "fix" won't cause a retain cycle somewhere, though it's enough of a workaround until an update of the SDK is out.

**Comment by chauduyphanvu** [comment]
Fixed in [v5.14.1](https://github.com/mopub/mopub-ios-sdk/commit/27c3bc5df7730079221edf887a17c2e35eed4a70). Let us know if you continue seeing the issue!
